### ISSUE TITLE
Cannot start imdocker module

### ISSUE THREAD

**Issue by nbjohnson** [rsyslog/rsyslog#5306]
### Expected behavior
`imdocker` module starts up and ingests docker info

### Actual behavior
Crashes on start with the error `main thread of imdocker terminated abnormally `

### Steps to reproduce the behavior
1. Install imdocker from `adiscon/v8-stable`
2. Docker installed on system, `/var/run/docker.sock` exists
3. Add new config to `/etc/rsyslog.d` with minimum config `module(load="imdocker")`
4. Restart rsyslog

### Environment
- rsyslog version: 8.2312.0
- platform: Ubuntu 20.04
- for configuration questions/issues, include rsyslog.conf and included config files
`module(load="imdocker")`

Not sure if I am not understanding the configuration required for imdocker or something else. Is there any place more detailed logs about the rsyslog startup failure can be found? The config validation didn't complain so rsyslog works fine without the imdocker module.

**Comment by rgerhards** [comment]
Please create a debug log. Instructions: https://www.rsyslog.com/doc/v8-stable/troubleshooting/howtodebug.html

**Be sure that the problem in question occurs at least once during the debug run**. Once this is done, please post the debug log (e.g. via pastebin or github). After that, we can review the contents and check how exactly rsyslog processing worked.

Note that imdocker is a contributed module. So it may take extra time to do troubleshooting. The debug log will show us valuable details in any case.

**Comment by n2yen** [comment]
hi @nbjohnson , if you could attach the items mentioned above, that would be super helpful!

**Comment by nbjohnson** [comment]
I was able to enable that debugging and parsed out some important logs:
```
imdocker.c listcontainers url: http/v1.27/containers/json?
imdocker.c getContainerIds - curl response: 7
errmsg.c Called LogMsg, msg main thread of imdocker terminated abnormally
iminternal.c signaling new iternal message via SIFTTOU main thread of imdocker terminated abnormally
```
Looks like the issue was rsyslog was not running as an elevated user so it did not have permission to access the docker socket, so running as elevated user or adding the syslog user to the docker group allowed it to connect successfully and pull data.

I am now working through pulling useful data and seem to be running into some limitations of the module, maybe I just havent dug deep enough into the configuration for the docker module, but seems like if a container is alive for a shorter time than the poll setting the logs are never pulled, so shorting the poll interval is an option but not sure the performance impact of that. Also the metadata for the image name seems to be the shortname and doesnt include the repo it was pulled from. Now that I actually have data just working through what options are available.

**Comment by Aaron2516** [comment]
> I was able to enable that debugging and parsed out some important logs:
> 
> ```
> imdocker.c listcontainers url: http/v1.27/containers/json?
> imdocker.c getContainerIds - curl response: 7
> errmsg.c Called LogMsg, msg main thread of imdocker terminated abnormally
> iminternal.c signaling new iternal message via SIFTTOU main thread of imdocker terminated abnormally
> ```
> 
> Looks like the issue was rsyslog was not running as an elevated user so it did not have permission to access the docker socket, so running as elevated user or adding the syslog user to the docker group allowed it to connect successfully and pull data.
> 
> I am now working through pulling useful data and seem to be running into some limitations of the module, maybe I just havent dug deep enough into the configuration for the docker module, but seems like if a container is alive for a shorter time than the poll setting the logs are never pulled, so shorting the poll interval is an option but not sure the performance impact of that. Also the metadata for the image name seems to be the shortname and doesnt include the repo it was pulled from. Now that I actually have data just working through what options are available.

I had the same problem, this worked for me

**Comment by rgerhards** [comment]
sry for loosing track but I just wanted to tell that I have worked on imdocker recently. I can also use it very well in our own containerization effort. I close this issue -feel free to re-open if the problem reoccurs.

New users: pls create a new issue, even if  you think it is familiar. A new issues provides more context for us. Thx!
